File: features/task/src/main/java/com/nivasafinance/features/task/exception/TaskNotFoundException.java
package com.nivasafinance.features.task.exception;

import com.nivasafinance.common.exception.ExceptionUtils;
import com.nivasafinance.common.exception.ResourceNotFoundException;
import org.springframework.context.MessageSource;

import java.io.Serial;
import java.util.UUID;

public class TaskNotFoundException extends ResourceNotFoundException {

    @Serial
    private static final long serialVersionUID = 1234567890123456789L;

    private TaskNotFoundException(String message) {
        super(message);
    }

    public static TaskNotFoundException taskNotFound(String taskIdentifier, MessageSource messageSource) {
        return new TaskNotFoundException(
            ExceptionUtils.createLocalizedMessage(
                "error.task.not.found",
                new Object[]{taskIdentifier},
                messageSource
            )
        );
    }

    public static TaskNotFoundException taskNotFoundByUuid(UUID taskUuid, MessageSource messageSource) {
        return new TaskNotFoundException(
            ExceptionUtils.createLocalizedMessage(
                "error.task.not.found",
                new Object[]{taskUuid},
                messageSource
            )
        );
    }
}

File: features/tasks/src/main/java/com/nivasafinance/features/tasks/exception/TaskNotFoundException.kt
package com.nivasafinance.features.tasks.exception

import com.nivasafinance.common.exception.ExceptionUtils
import com.nivasafinance.common.exception.ResourceNotFoundException
import org.springframework.context.MessageSource
import java.util.UUID

class TaskNotFoundException(
    taskId: UUID,
    messageSource: MessageSource
) : ResourceNotFoundException(
    ExceptionUtils.createLocalizedMessage(
        "error.task.not.found",
        arrayOf(taskId.toString()),
        messageSource
    )
)
